core/src/main/java/io/grpc/internal/ServerImpl.java

```diff
@@ -81,7 +81,7 @@
 
   /** Executor for application processing. */
   private Executor executor;
-  @GuardedBy("lock") private boolean usingSharedExecutor;
+  private boolean usingSharedExecutor;
```

Comment:
It must be set before `transportServer.start()` (which means it couldn't be removed from the synchronized block even if it didn't need it) and it needs some coordination with shutdown(). It needs synchronization, but isn't the GuardedBy semantics.